File: go/word-count/word_count.go
// Package wordcount provides Word Count solution.
package wordcount

import (
	"regexp"
	"strings"
)

// Frequency is word frequency in a phrase.
type Frequency map[string]int

var wordRegex = regexp.MustCompile(`\b[\w']+\b`)

// WordCount returns the word frequency of s.
func WordCount(s string) Frequency {
	words := wordRegex.FindAllString(s, -1)
	result := make(Frequency)
	for _, word := range words {
		result[strings.ToLower(word)]++
	}
	return result
}

File: ruby/word-count/word_count.rb
class Phrase
  attr_reader :word_count

  def initialize(phrase)
    count_by_word = Hash.new(0)
    phrase.downcase.scan(/\b[\w']+\b/).each do |word|
      count_by_word[word] += 1
    end
    @word_count = count_by_word
  end
end
